core/src/test/scala/org/apache/spark/HeartbeatReceiverSuite.scala

```diff
@@ -225,6 +270,7 @@ class HeartbeatReceiverSuite
         Matchers.eq(Array(1L -> metrics.accumulators())),
         Matchers.eq(blockManagerId))
     }
+    expectedUpdatedEpoch.foreach{_ => assert(response.updatedEpoch === expectedUpdatedEpoch)}
```

Comment:
don't you want to do this assert even if the expectedUpdate is `None`?

Reply:
you are right, I was only thinking on the positive case